manage.py

```diff
@@ -190,6 +217,346 @@ def _add_validate_schema_subparser(subparsers):
+    )
+
+    # Set the name of this parser to 'metrics'
+    metrics_parser.set_defaults(command='metrics')
+
+    # allow the user to select 1 or more functions to enable metrics for
+    metrics_parser.add_argument(
+        '-f', '--functions',
+        choices=['rule', 'alert', 'athena'],
+        help=ARGPARSE_SUPPRESS,
+        nargs='+',
+        action=NormalizeFunctionAction,
+        required=True
+    )
+
+    # get the metric toggle value
+    toggle_group = metrics_parser.add_mutually_exclusive_group(required=True)
+
+    toggle_group.add_argument(
+        '-e', '--enable',
+        dest='enable_metrics',
+        action='store_true'
+    )
+
+    toggle_group.add_argument(
+        '-d', '--disable',
+        dest='enable_metrics',
+        action='store_false'
+    )
+
+    # allow the user to select 0 or more clusters to enable metrics for
+    metrics_parser.add_argument(
+        '-c', '--clusters',
+        choices=clusters,
+        help=ARGPARSE_SUPPRESS,
+        nargs='+',
+        action=UniqueSetAction,
+        default=clusters
+    )
+
+    # allow verbose output for the CLI with the --debug option
+    metrics_parser.add_argument(
+        '--debug',
+        action='store_true',
+        help=ARGPARSE_SUPPRESS
+    )
+
+
+def _add_metric_alarm_subparser(subparsers):
+    """Add the create-alarm subparser: manage.py create-alarm [options]"""
+    metric_alarm_usage = 'manage.py create-alarm [options]'
+
+    # get the available metrics to be used
+    available_metrics = metrics.MetricLogger.get_available_metrics()
+    all_metrics = [metric for func in available_metrics for metric in available_metrics[func]]
+
+    metric_choices_block = ('\n').join('{:>35}{}'.format('', metric) for metric in all_metrics)
+
+    # get cluster choices from available files
+    clusters = [os.path.splitext(cluster)[0] for _, _, files
+                in os.walk('conf/clusters') for cluster in files]
+
+    cluster_choices_block = ('\n').join('{:>37}{}'.format('', cluster) for cluster in clusters)
+
+    metric_alarm_description = ("""
+StreamAlertCLI v{}
+Add a CloudWatch alarm for predefined metrics. These are save in the config and
+Terraform is used to create the alarms.
+
+Required Arguments:
+
+    -m/--metric                  The predefined metric to assign this alarm to. Choices are:
+{}
+    -mt/--metric-target          The target of this metric alarm, meaning either the cluster metric
+                                   or the aggrea metric. Choices are:
+                                     cluster
+                                     aggregate
+                                     all
+    -co/--comparison-operator    Comparison operator to use for this metric. Choices are:
+                                   GreaterThanOrEqualToThreshold
+                                   GreaterThanThreshold
+                                   LessThanThreshold
+                                   LessThanOrEqualToThreshold
```

Comment:
It would be more user-friendly if you allow `>=, >, <, <=` and convert to the longer form, like you do for the function names

Reply:
That's smart - I was just using aws terms as a template but I like that approach better

Reply:
On second thought I think I'll leave it as is - using `$ python manage.py create-alarm --help`  will show the acceptable options.

Also if an invalid param is used, it will print the options like: `manage.py [command] [subcommand] [options] create-alarm: error: argument -co/--comparison-operator: invalid choice: 'GreaterThanOrEqualToThresho' (choose from 'GreaterThanOrEqualToThreshold', 'GreaterThanThreshold', 'LessThanThreshold', 'LessThanOrEqualToThreshold')`

A good thought but I think I'd just like to mirror the AWS options :)